trlx/model/accelerate_base_model.py

```diff
@@ -44,7 +45,13 @@ def __init__(self, config, train_mode=True):
         self.max_length = config.train.seq_length
 
         # Retrieves model equipped for ppo, ilql, etc
-        self.model = self.get_arch(self.config)
+        torch_dtype = {
+            "float32": torch.float32,
+            "float16": torch.float16,
+            "bfloat16": torch.bfloat16
+        }[self.config.model.dtype]
```

Comment:
This might be redundant as `accelerate` allows users to specify half floating-point types in the `mixed_precision` config. Have you tried those settings?

Reply:
Have not, seems obvious in hindsight, was writing a bit fast